### ISSUE TITLE
Adding Teams to Clubs

### ISSUE THREAD

**Issue by seanoshea** [rgbycch/rgbycch-rest#19]
Should be able to do this via the api. Similar to players and teams.

**Comment by gffny** [comment]
I'll take this on tonight. Picked up an injury yesterday, so there'll be plenty of sitting for a while

**Comment by seanoshea** [comment]
@gffny check out https://github.com/rgbycch/rgbycch_swift_api/commit/5ffec92a1e2af5bb552dd0bbc9ef490fd24aa5ac for where I was thinking of going with this.

**Comment by gffny** [comment]
@seanoshea the url matches what I have done for the 'add team'. I have to do the same for 'remove team'. Quick question, in this context, is team ~= a match day panel or a divisional teams within a club ~= Blackrock 1st VX, Blackrock 2nd XV, etc.?

**Comment by seanoshea** [comment]
@gffny - Good question there. I had assumed a team was more like a Blackrock 1st XV, Blackrock 2nd XV. A team could have anything up to 30 players. Whereas a matchday squad would have a max of 23 (I think you're only allowed to name 8 subs)

**Comment by seanoshea** [comment]
FWIW, I think something similar to the `add_player` and `remove_player` routes would work. i.e. in `routes.rb` we have:

```
      resources :teams, only: [:show, :create, :update, :destroy, :index] do
        put 'add_player'
        put 'remove_player'
      end
```

maybe something like:

```
      resources :clubs, only: [:show, :create, :update, :destroy, :index] do
        put 'add_team'
        put 'remove_team'
      end
```

would work? I haven't tested this out - just an idea.

**Comment by gffny** [comment]
Cool. We're on the same page about Teams. The route I set up is working but it doesn't return a JSON representation of the club upon conpletition

**Comment by seanoshea** [comment]
yeah. there's an issue with serializers and circular references (clubs have teams and teams have clubs) which means that you need to double check the clubs/teams serializers to make sure they're ok.

**Comment by seanoshea** [comment]
@gffny Check out https://github.com/rgbycch/rgbycch_swift_api/pull/13 for the client side work for this.
